romeo/asset-contract/contracts/asset.clar

```diff
@@ -0,0 +1,94 @@
+;; title: asset
+;; version:
+;; summary: sBTC dev release asset contract
+;; description:
+
+;; traits
+;;
+(impl-trait 'ST1NXBK3K5YYMD6FD41MVNP3JS1GABZ8TRVX023PT.sip-010-trait-ft-standard.sip-010-trait)
+
+;; token definitions
+;;
+(define-fungible-token sbtc u21000000)
+
+;; constants
+;;
+(define-constant err-invalid-caller u1)
+(define-constant err-not-token-owner u2)
+
+;; data vars
+;;
+(define-data-var contract-owner principal tx-sender)
+(define-data-var bitcoin-wallet-public-key (optional (buff 33)) none)
+
+;; public functions
+;;
+(define-public (set-bitcoin-wallet-public-key (public-key (buff 33)))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set bitcoin-wallet-public-key (some public-key)))
+    )
+)
+
+(define-public (mint! (amount uint) (dst principal) (deposit-txid (string-ascii 72)))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        ;; TODO #79: Assert deposit-txid exists on chain
```

Comment:
I would ping Setzeus about this cause am pretty sure this logic already exists in the mini contract.

Reply:
True, although Mini does complete validation of the bitcoin transaction and all we need here is to assert that it exists.

Involving Setzeus here makes sense either way. Just noticed I can't ping him. I'll request to add him to the repo.